### ISSUE TITLE
ARG002 on callbacks

### ISSUE THREAD

**Issue by matthiasverstraete** [astral-sh/ruff#10409]
Hi,

I'm not sure if this is the right place to post this question, so please redirect me if necessary.
Below is a code example:

some_library.py
```python
def foo(data: List[int], callback: Callable[[str, int], int]) -> None:
    total = 0
    for d in data:
        total += callback(str(d), d)
```

my_code.py
```python
from some_library import foo

def my_callback(name: str, value: int) -> int:
    return value * 2

foo([1,2,3,4], my_callback)
```

In this example, the rule ARG002 is triggered because I'm not using `name` in my callback. However, I can't remove that argument or mypy will complain that the type doesn't match.

In this situation, is the only solution to add `# noqa: ARG002`, or is there a proper way to handle this?

Thanks for your input!

**Comment by charliermarsh** [comment]
You could change `name: str` to `_name: str`, but that also carries some risk, since it will break for users that pass keyword arguments. In general, yeah, I think a `# noqa` is required here if you're trying to adhere to some informal but common interface.

**Comment by charliermarsh** [comment]
The other thing you could do is make it an abstract class so that there's a formal interface. Then, if you decorate the method with `@override`, Ruff will correctly detect that the args are required.

**Comment by charliermarsh** [comment]
Hope that's helpful!

**Comment by pianofab** [comment]
Pylint handles this automatically by recognizing certain prefixes in the function name (_cb and such), avoiding littering one’s code with noqa statements. Callbacks are extremely common for example in WxPython code. It would be great to handle it just like Pylint.
